mgv2/mgv2.py

```diff
@@ -0,0 +1,307 @@
+
+
+        connection = redis.Redis(connection_pool=Mgv2.pool)
+
+        model_info = connection.get(json.dumps(lost_ip))
+
+        keys = connection.keys('*')
+        for key in keys:
+            logger.info("Key is {} and value is {}".format(key, connection.get(key)))
+
+        connection.set(json.dumps(lost_ip), json.dumps([]))
+
+        model_info_lst = json.loads(model_info)
+
+        logger.info("We are now going to redeploy {} models".format(len(model_info_lst)))
+
+        for x in range(len(model_info_lst)):
+            m = model_info_lst[x]
+
+            Mgv2.container_manager.deploy_model(m["name"], m["version"], m["input_type"], m["image"], 1, new_ip)
+
+        # with Mgv2.database_lock.get_lock():
+        #     Mgv2.database_lock.value = 0
+
+
+
+        dead = connection.get("dead_ips")
+        if dead:
+            value = json.loads(dead)
+            if [lost_ip, new_ip] not in value:
+                value.append([lost_ip, new_ip])
+                value = json.dumps(value)
+                connection.set("dead_ips", value)
+        else:
+            value = json.dumps([[lost_ip, new_ip]])
+            connection.set("dead_ips", value)
+
+
+
+    def dbg(self):
+        last_time = time.time()
+
+        while True:
+            now = time.time() - last_time
+            # print("Debug; Time is now {}", now)
+            print("value of boolean is {}", Mgv2.database_lock.value)
+            time.sleep(1)
+
+    @staticmethod
+    @app.route('/begin_polling', methods=['GET', 'POST'])
+    def begin_polling():
+        logger.info("Signal to start polling has been received")
+
+        try:
+            data = request.get_json(force = True)
+            logger.info("Received data is {}".format(data))
+            type = json.loads(data)['type']
+            if type == 'kubernetes':
+                Mgv2.container_manager = KubernetesContainerManager(kubernetes_api_ip="eric-dev.clipper-k8s-dev.com",
+                                                                    useInternalIP=True)
+            elif type == 'docker':
+                Mgv2.container_manager = DockerContainerManager()
+
+            Mgv2.container_manager.connect()
+
+            logger.info("Starting polling")
+            polling = Process(target=Mgv2.poll_forever)
+            polling.start()
+            return 'Success'
+        except Exception as e:
+            traceback.print_exc()
+            return 'Failure'
+
+    @staticmethod
+    @app.route('/get_query_to_model_mapping', methods=['GET', 'POST'])
+    def get_query_to_model_mapping():
+
+
+        logger.info("Flask has received model info")
+
+        try:
+            query_to_model_mapping = request.get_json(force = True)
+
+
+            if not isinstance(query_to_model_mapping, dict):
+                query_to_model_mapping = json.loads(query_to_model_mapping)
+
+            logger.info("Query_to_model_mapping: {}".format(query_to_model_mapping))
```

Comment:
Refactor the rest of this method into a separate method that takes the `query_to_model_mapping` dict. Then you can both call it here, and after you call deploy_model in `reassign...`